sng_freepbx_debian_install.sh

```diff
@@ -94,24 +113,11 @@ isinstalled() {
 	fi
 }
 
-# Function to install the package 
+# Function to install the package
 pkg_install() {
-	log "############################### "
 	PKG=$@
-	if isinstalled $PKG; then
-		log "$PKG already present ...."
-	else 
-		message "Installing $PKG Now ...."
-		apt-get -y --ignore-missing -o DPkg::Options::="--force-confnew" -o Dpkg::Options::="--force-overwrite" install $PKG >> $log 2>&1
-		if isinstalled $PKG; then
-			message "$PKG installed successfully...."
-		else 
-			message "$PKG failed to install ...."
-			message "Exiting the installation process as dependent $PKG failed to install ...."
-			terminate
-		fi
-	fi
-	log "############################### "
+	log "Installing $PKG"
+	apt -y -o DPkg::Options::="--force-confnew" -o Dpkg::Options::="--force-overwrite" install $PKG >> $log 2>&1
```

Comment:
Hi @tdltdl If the package is already present then no need to try to install it again, there is no harm but above logic is also checking after apt install package actually getting installed or not and if it failed to install due to any reason then we are exiting from the script, so i think no need to change pkg_install function.

Reply:
Hi @kguptasangoma Why do we need to check if the package is already installed? Why don't we trust apt to do his job?
I still believe we should not have 
- force-confnew because if I start from a template VM where everything is already pre-configured like openSSL with enforced security, mail server already configured to send mails and so on, the last thing I want is that an installation screw all this for no reason
- force-overwrite is really odd (except if you are afraid someone played with your binaries
and worst of all, all those predicates will be ignored as if the package is already installed, it will not be called ;-)

If apt install <list of packages> fails, the script will also stop as apt will return a non-zero code(this is the point of the set -e)

So I really believe we should just replace this by "apt install -y $PKG" >>$log 2>&1
The other advantage is that speed will be dramatically improved (for each packages today, the script is 

1. checking if the package is installed
2. install the package that requires checking if it is already installed, check all dependencies, add them f needed to the list of packages to install and finally installing it, updating the package list.

Calling it with th whole list (even if they are already installed) means that all this dependency checks are done only once (not mentioning that if you have security packages like rkhunter, after each execution of apt install it scans the system again to update metadata...)

If really you want to start from clean config for, let's say asterisk and freepbx, I would recommend you do a apt purge of all those packages and their dependencies before starting the install and rely on apt to decide when a new config needs to be installed.

Also I do believe as part of the "make sure env is sane" phase, we should do an apt upgrade. This will ensures that we have all packages (and dependencies) updated, avoiding potential issues with some old version installed that are not compatible with the new packages being installed.